pkg/controller/controller_binding.go

```diff
@@ -400,7 +400,24 @@ func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[st
 		},
 	}
 
-	_, err = c.kubeClient.SettingsV1alpha1().PodPresets(binding.Namespace).Create(podPreset)
+	found = false
+	_, err = c.kubeClient.SettingsV1alpha1().PodPresets(binding.Namespace).Get(podPreset.ObjectMeta.Name, metav1.GetOptions{})
+	if err == nil {
+		found = true
+	}
+
+	if found {
+		// TODO: Check ownerReferences first?
+		// NOTE: PodPresets don't support Update requests
+		err = c.kubeClient.SettingsV1alpha1().PodPresets(binding.Namespace).Delete(podPreset.ObjectMeta.Name, &metav1.DeleteOptions{})
```

Comment:
This should be a conditional delete with UID precondition - otherwise you may end up deleting an object created by a concurrently running controller. ABA problem.